api/rdb/v1/rdb_utils.go

```diff
@@ -0,0 +1,49 @@
+package rdb
+
+import (
+	"time"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/async"
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// WaitForInstanceRequest is used by WaitForInstance method.
+type WaitForInstanceRequest struct {
+	InstanceID string
+	Region     scw.Region
+	Timeout    time.Duration
+}
+
+// WaitForInstance waits for the instance to be in a "terminal state" before returning.
+// This function can be used to wait for an instance to be ready for example.
+func (s *API) WaitForInstance(req *WaitForInstanceRequest) (*Instance, error) {
+
+	terminalStatus := map[InstanceStatus]struct{}{
+		InstanceStatusReady:    {},
+		InstanceStatusDiskFull: {},
+		InstanceStatusError:    {},
+	}
+
+	instance, err := async.WaitSync(&async.WaitSyncConfig{
+		Get: func() (interface{}, error, bool) {
+			res, err := s.GetInstance(&GetInstanceRequest{
+				InstanceID: req.InstanceID,
+				Region:     req.Region,
+			})
+
+			if err != nil {
+				return nil, err, false
+			}
+			_, isTerminal := terminalStatus[res.Status]
+
+			return res, nil, isTerminal
+		},
+		Timeout:          req.Timeout,
+		IntervalStrategy: async.LinearIntervalStrategy(5 * time.Second),
```

Comment:
IMO, this should be a constant we could reuse for other `WaitFor..()` func

Reply:
This constant would have to live in `scw.WaitForConstant` to avoid cross-product imports, let's keep the product logic inside the product namespace.

Reply:
We'll only have 1 `WaitFor..` func per product ? 

Should settings like timeouts or intervals really live inside a product ? Don't we want some consistency across all products ? 

What is the difference with rate limiting ?